journal.py

```diff
@@ -0,0 +1,212 @@
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from pyramid.security import remember, forget
+
+Base = declarative_base()
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://jimgrant@localhost:5432/learning-journal'
+)
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
+
+    def __repr__(self):
+        return self.title
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def one(cls, eid=None, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter(cls.id == eid).one()
+
+    @classmethod
+    def modify(cls, eid=None, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls.one(eid)
+        instance.title = title
+        instance.text = text
+        session.add(instance)
+        return instance
+
+
+def init_db():
+    engine = sa.create_engine(DATABASE_URL)
+    Base.metadata.create_all(engine)
+
+
+@view_config(route_name='home', renderer='templates/list.jinja2')
+def list_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def detail_view(request):
+    entry = Entry.one(request.matchdict['id'])
+    return {'entry': entry}
+
+
+@view_config(route_name='create', renderer='templates/create.jinja2')
+def create_view(request):
+    return {}
+
+
+@view_config(route_name='edit', renderer='templates/edit.jinja2')
+def edit_view(request):
+    entry = Entry.one(request.matchdict['id'])
+    return {'entry': entry}
+
+
+@view_config(route_name='add', request_method='POST')
+def add_entry(request):
+    title = request.params.get('title')
+    text = request.params.get('text')
+    Entry.write(title=title, text=text)
+    return HTTPFound(request.route_url('home'))
+
+
+@view_config(route_name='modify', request_method='POST')
+def modify_entry(request):
+    eid = request.matchdict['id']
+    title = request.params.get('title')
+    text = request.params.get('text')
+    Entry.modify(eid=eid, title=title, text=text)
+    return HTTPFound(request.route_url('home'))
```

Comment:
dang, man.  I'm going to have to come up with something else for you to do this week!